File: app/src/main/java/ru/nehodov/weatherforecast/dao/CurrentDao.java
package ru.nehodov.weatherforecast.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import ru.nehodov.weatherforecast.entities.Daily;

@Dao
public interface CurrentDao {

    @Insert
    void insertCurrentWeather(Daily daily);

    @Query("SELECT * FROM current")
    LiveData<Daily> getCurrentWeather();

    @Query("DELETE FROM current")
    void deleteCurrentWeather();
}

File: app/src/main/java/ru/nehodov/weatherforecast/dao/CurrentDao.kt
package ru.nehodov.weatherforecast.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import kotlinx.coroutines.flow.Flow
import ru.nehodov.weatherforecast.entities.Current

@Dao
interface CurrentDao {

    @Insert
    suspend fun insert(daily: Current)

    @Query("SELECT * FROM current")
    fun currentWeatherData(): Flow<Current>

    @Query("DELETE FROM current")
    suspend fun deleteAll()
}